packages/x-charts-pro/src/ChartOverview/ChartAxisZoomOverview.tsx

```diff
@@ -0,0 +1,330 @@
+      // Prevent text selection when dragging the handle
+      event.preventDefault();
+      prev = axisDirection === 'x' ? event.clientX : event.clientY;
+      document.addEventListener('pointerup', onPointerUp);
+      document.addEventListener('pointermove', onPointerMove);
+    };
+
+    activePreviewRect.addEventListener('pointerdown', onPointerDown);
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      activePreviewRect.removeEventListener('pointerdown', onPointerDown);
+      onPointerMove.clear();
+    };
+  }, [axisDirection, axisId, instance, store]);
+
+  const onResizeStart = (delta: number) => {
+    store.update((state) => {
+      const { width, height } = selectorChartDrawingArea(state);
+      const drawingAreaSize = axisDirection === 'x' ? width : height;
+
+      const zoomOptions = selectorChartAxisZoomOptionsLookup(state, axisId);
+
+      const newState = {
+        ...state,
+        zoom: {
+          ...state.zoom,
+          zoomData: state.zoom.zoomData.map((zoom) => {
+            if (zoom.axisId === axisId) {
+              const deltaZoom = (delta / drawingAreaSize) * 100;
+
+              return {
+                ...zoom,
+                start: calculateZoomStart(zoom.start + deltaZoom, zoom, zoomOptions),
+              };
+            }
+
+            return zoom;
+          }),
+        },
+      };
+
+      return newState;
+    });
+  };
+
+  const onResizeEnd = (delta: number) => {
+    store.update((state) => {
+      const { width, height } = selectorChartDrawingArea(state);
+      const drawingAreaSize = axisDirection === 'x' ? width : height;
+
+      // TODO: What about non-cartesian axes? Are these the only ones that can be zoomed?
+      const zoomOptions = selectorChartAxisZoomOptionsLookup(state, axisId);
+
+      const newState = {
+        ...state,
+        zoom: {
+          ...state.zoom,
+          zoomData: state.zoom.zoomData.map((zoom) => {
+            if (zoom.axisId === axisId) {
+              const deltaZoom = (delta / drawingAreaSize) * 100;
+
+              return {
+                ...zoom,
+                end: calculateZoomEnd(zoom.end + deltaZoom, zoom, zoomOptions),
+              };
+            }
+
+            return zoom;
+          }),
+        },
+      };
+
+      return newState;
+    });
+  };
+
+  return (
+    <React.Fragment>
+      <ZoomRangePreviewRect
+        ref={activePreviewRectRef}
+        x={axisDirection === 'x' ? (zoomData.start / 100) * drawingArea.width : 0}
+        y={axisDirection === 'x' ? 0 : (zoomData.start / 100) * drawingArea.height}
+        width={
+          axisDirection === 'x' ? (drawingArea.width * (zoomData.end - zoomData.start)) / 100 : size
+        }
+        height={
+          axisDirection === 'x'
+            ? size
+            : (drawingArea.height * (zoomData.end - zoomData.start)) / 100
+        }
+      />
+      {
+        // TODO: In RTL languages, should we start from the right?
+      }
```

Comment:
You have `axis.reverse` than might impact this. ANd we coudl consider that RtL is "just" modifying this value if needed

Reply:
I added support for `reverse` in this PR, but I'll add RTL in a follow-up because from my tests the axis don't seem to be respecting RTL (maybe I did something wrong).

Reply:
Hum, I don't think so. Though I have no experience with RTL languages... 😅 

This is my understanding of our current implementation of RTL charts, where we only change the text direction. 

An easy way to think about RTL is that in LTR, the time passes to the right `x -> y`. While on RTL languages, it is the other way around `y <- x`. 

This would mean that in some instances, when the X axis depicts time or a progressive series, it should probably be reverted, while in other cases, like categorical, it doesn't matter.

We end up leaving this to the user, and only changing the text direction as that is always desirable in our case, but we don't change other behaviours, as they are situanional.